lib/seed/facility_seeder.rb

```diff
@@ -123,17 +119,25 @@ def create_district_regions(state_results, facility_group_results)
     end
 
     def create_block_regions(district_region_results)
-      block_count = district_region_results.ids.size
+      # Eagerly fetch block names to avoid duplicates
+      block_count = district_region_results.ids.size * max_number_of_blocks_per_facility_group
       block_names = Seed::FakeNames.instance.blocks.sample(block_count)
-      block_regions = district_region_results.results.each_with_index.map { |row, i|
+
+      block_counter = 0
```

Comment:
This took a while to understand 😅 , moving the sampling in should simplify (at the expense of fully distinct names).

Reply:
Yup indeed. I'll do a test with a handful of block names and tons of duplication, and see if our reports still hold up.

Reply:
Ah I've found the problem. It's not blocks with the same name, rather - the ActiveRecord import that we do isn't smart enough to assign unique slugs to blocks with the same name. I'm going to leave this as is.

Reply:
This is where the the slug generation happens, it should work out as long as they have a different uuid 🤔  https://github.com/simpledotorg/simple-server/blob/71db34c7219508ae4422db530d5e3b60874de457/app/models/region.rb#L89

Reply:
Nah, ActiveRecord import bypasses these application-level hooks.

Reply:
Ah ok 👍🏼 . Non-blocking: Do you know where the slugs get generated in that case?

Reply:
That's a good point. Clearly activerecord import does some weird middle ground where slug generation is done, but uniqueness isn't checked? Might make sense, right? The slug candidates are chosen based on records already persisted in the DB. When you're importing, nothing's persisted yet. 🤷 

It's weird enough that I'm gonna leave it alone for now 😅